release/downloadIstioCtl.sh

```diff
@@ -43,22 +44,64 @@ if [ "x${ISTIO_VERSION}" = "x" ] ; then
   exit;
 fi
 
+LOCAL_ARCH=$(uname -m)
+if [ "${TARGET_ARCH}" ]; then
+    LOCAL_ARCH=${TARGET_ARCH}
+fi
+
+case "${LOCAL_ARCH}" in 
+  x86_64)
+    ISTIO_ARCH=amd64
+    ;;
+  armv8*)
+    ISTIO_ARCH=arm64
+    ;;
+  aarch64*)
+    ISTIO_ARCH=arm64
+    ;;
+  armv*)
+    ISTIO_ARCH=armv7
+    ;;
+  amd64|arm64)
+    ISTIO_ARCH=${LOCAL_ARCH}
+    ;;
+  *)
+    echo "This system's architecture, ${LOCAL_ARCH}, isn't supported"
+    exit 1
+    ;;
+esac
+
 download_failed () {
   printf "Download failed, please make sure your ISTIO_VERSION is correct and verify the download URL exists!"
   exit 1
 }
 
 # Downloads the istioctl binary archive.
 tmp=$(mktemp -d /tmp/istioctl.XXXXXX)
-filename="istioctl-${ISTIO_VERSION}-${OSEXT}.tar.gz"
+NAME="istioctl-${ISTIO_VERSION}"
+
 cd "$tmp" || exit
 URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istioctl-${ISTIO_VERSION}-${OSEXT}.tar.gz"
-printf "Downloading %s from %s ... \n" "${filename}" "${URL}"
-curl -sLO "${URL}" || download_failed
+ARCH_URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istioctl-${ISTIO_VERSION}-${OSEXT}-${ISTIO_ARCH}.tar.gz"
+
+printf "Downloading %s from %s ... \n" "${NAME}" "${URL}"
+if ! curl -fsLO "$URL"
+then
+  printf "Failed. \n\nTrying with ISTIO_ARCH. Downloading %s from %s ...\n" "${NAME}" "$ARCH_URL"
```

Comment:
I don't think ISTIO_ARCH is correct as someone really can't set that value prior to calling the script, as it just gets set. You have TARGET_ARCH in the printf in the other file which is better. At least if we document that it's the `unmet -m` output that we are looking for (as opposed to amd64).

Reply:
Good catch. I fixed in the other file. Missed this one.

Reply:
From the user, the script will accept either x86_64, armv8, aarch64, armv, amd64, or arm64. Uname -m or common name for the arch :)